fedimint-core/src/epoch.rs

```diff
@@ -12,14 +12,29 @@ use threshold_crypto::{PublicKey, PublicKeySet, Signature, SignatureShare};
 
 use crate::transaction::Transaction;
 
+/// All the items that may be produced during a consensus epoch
 #[derive(Debug, Clone, Eq, PartialEq, Hash, UnzipConsensus, Encodable, Decodable)]
 pub enum ConsensusItem {
+    /// Fed shutdown occurs once a threshold of peers signal (for consensus
+    /// upgrades)
+    ShutdownSignal(ShutdownReason),
+    /// Threshold sign the configs for verification via the API
     ClientConfigSignatureShare(SerdeSignatureShare),
+    /// Threshold sign the epoch history for verification via the API
     EpochOutcomeSignatureShare(SerdeSignatureShare),
+    /// Fedimint tx that can contain module inputs and outputs that must be net
+    /// equal
     Transaction(Transaction),
+    /// Any data that modules require consensus on
     Module(ModuleConsensusItem),
 }
 
+/// Contains consensus info about the shutdown reason
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Encodable, Decodable)]
+pub enum ShutdownReason {
+    Upgrade,
+}
+
```

Comment:
Agree with DPC for now an enum with 1 variant is kinda unnecessary. In terms of upgradeability it also doesn't add much without `#[non_exhaustive]` and afaik `ShutdownReason` shouldn't be part of the public API anyway? I'd say leave it out ofr now and add it if we ever have another reason.